src/snowflake/connector/aio/_connection.py

```diff
@@ -87,7 +88,19 @@ def __init__(
         # get the imported modules from sys.modules
         # self._log_telemetry_imported_packages() # TODO: async telemetry support
         # check SNOW-1218851 for long term improvement plan to refactor ocsp code
-        # atexit.register(self._close_at_exit) # TODO: async atexit support/test
+        atexit.register(self._close_at_exit)
+
+    def __enter__(self):
+        # async connection does not support sync context manager
+        raise TypeError(
+            "'SnowflakeConnection' object does not support the context manager protocol"
```

Comment:
shoud we raise `NotImplementedError` instead?

Reply:
this is to align the native python error experience if a class does not support context manager